src/plugins/auto/auto_schedule.cpp

```diff
@@ -79,25 +79,75 @@ void AutoSchedule::GenerateWorkers(const std::string& device,
                 IdleGuard<NotBusyPriorityWorkerRequests> idleGuard{workerRequestPtr, *idleWorkerRequestsPtr};
                 workerRequestPtr->_exceptionPtr = exceptionPtr;
                 {
-                    auto capturedTask = std::move(workerRequestPtr->_task);
-                    capturedTask();
-                }
-                // try to return the request to the idle list (fails if the overall object destruction has began)
-                if (idleGuard.Release()->try_push(std::make_pair(workerRequestPtr->_index, workerRequestPtr))) {
-                    // let's try to pop a task, as we know there is at least one idle request, schedule if succeeded
-                    // if no device-agnostic tasks, let's try pop the device specific task, schedule if succeeded
-                    IE::Task t;
-                    do {
-                        _inferPipelineTasks.try_pop(t);
-                    } while (t && ScheduleToWorkerInferRequest(std::move(t)));
-                    do {
-                        _inferPipelineTasksDeviceSpecific[device]->try_pop(t);
-                    } while (t && ScheduleToWorkerInferRequest(std::move(t), device));
+                    if (workerRequestPtr->_exceptionPtr != nullptr) {
+                        if (workerRequestPtr->_testExec->count == 1) {
+                            auto capturedTask = std::move(workerRequestPtr->_task);
+                            capturedTask();
+                        } else {
+                            workerRequestPtr->_reload = true;
+                            auto capturedTask = std::move(workerRequestPtr->_testExec->_task);
+                            workerRequestPtr->_testExec->count++;
+                            selectOtherDevice(workerRequestPtr->_deviceName);
+                            if (_loadContext[FALLBACKDEVICE].isAlready) {
+                                workerRequestPtr->_startTimes.pop_back();
```

Comment:
why remove starttime here

Reply:
Because if reload success, the original request will only has start time, it will assert in https://github.com/openvinotoolkit/openvino/blob/0860db0dc3e366e6cb54c03943de1ef3710a9afa/src/plugins/auto/multi_schedule.cpp#L241

Reply:
then I think we should fill in a endtime properly for this infer request, isn't it?

Reply:
Agree